README.md

```diff
@@ -0,0 +1,90 @@
+ - View: The view in our implementation of MVVM is actually a **Fragment or an Activity**. The views enclose everything needed to **handle the user interface**. **They observe the ViewModel**, using LiveData components, **and react to its changes as they need to**.
+ - ViewModel: The **responsable of the relationship between the data and the user interface of the application**. The ViewModel **has the logic to convert what the use cases provide into information that the view can understand and present**. Furthermore, it **has the responsability and the logic to react to the user’s input, and call the necessary use cases**. The most useful of the Android’s ViewModel class is its lifecycle consciousness. It only **communicates to the View with LiveData components, so it does not need a reference to a contexts or an activity**: it can keep the information alive even against configuration changes like screen rotations or calls to background.
+
+<p align="center">
+  <img src="https://koenig-media.raywenderlich.com/uploads/2019/02/MVVM.png" width="500" title="MVVM diagram"/>
+</p>
+
+If you want to read and understad at all MVVM, just visit: https://docs.google.com/presentation/d/1JkepsjABKhOmQr7bc62mL8vd-3UqbyklceHfJn4K0rU/edit?ts=5daa0000#slide=id.g61baf37ef4_1_33
+
+### Clean Architecture:
+Clean architecture attempts to **provide a methodology to use in coding in order to make it easier to develop quality code that performs better is easier to change, update and has fewer dependencies**.
+An important goal of clean architecture is to provide developers with a way to **organize code in such a way that it encapsulates the business logic but keeps it separate from the delivery mechanism**. 
+
+Layers of Clean Architecture used in this project:
+ - Domain: Would execute **business logic which is independent of any layer and is just a pure kotlin package with no android specific dependency**.
+ - Data: Would **dispense the required data** for the application to the domain layer by **implementing interface exposed by the domain**.
+ - Presentation: Would **include both domain and data layer** and is android specific which **executes the UI logic**.
+ 
+ <p align="center">
+  <img src="https://miro.medium.com/max/700/1*jH0iI7-MSQYgLUrqTUm6mg.png" width="500" title="clean architecture layer rings">
+</p>
+
+To learn about different ways of implement Clean Architecture, visit: https://docs.google.com/presentation/d/1aBtccGraTyggnIP6Nn7m8uGfBgreKWIk-2JuLafKAds/edit#slide=id.p4
+
+### Dependency Injection - Koin
+ - Dependency injection is a programming **technique that makes a class independent of its dependencies by decoupling the usage of an object from its creation**. It’s derived from the fifth principle of the famous object oriented programming principles S.O.L.I.D (class should concentrate on fulfilling its own responsibilities and should not be concerned with creating objects to fulfill those responsibilities).
```

Comment:
Be careful with your explanation of Dependency Injection because you say that it comes from the D of SOLID and the D of solid is not Dependency injection, it is Dependency inversion

Reply:
Yes I did not want to say that the D of SOLID was Dependency Injection, refactored!